src/main/scripts/run_e2e_test_module.py

```diff
@@ -0,0 +1,110 @@
+
+def run_shell_command(cmd):
+    process = subprocess.run(cmd.split(" "))
+    assert process.returncode == 0
+
+if __name__ == "__main__":
+    # Start CDAP sandbox
+    print("Downloading CDAP sandbox")
+    sandbox_url = "https://builds.cask.co/artifact/CDAP-BUT/shared/build-latestSuccessful/SDK/cdap/cdap-standalone/target/cdap-sandbox-6.8.0-SNAPSHOT.zip"
+    sandbox_dir = sandbox_url.split("/")[-1].split(".zip")[0]
+    r = requests.get(sandbox_url)
+    z = zipfile.ZipFile(io.BytesIO(r.content))
+    z.extractall("./sandbox")
+    print("Start the sandbox")
+    run_shell_command(f"chmod +x sandbox/{sandbox_dir}/bin/cdap")
+    os.system("export JAVA_OPTS=-Xmx16G")
+    run_shell_command(f"sandbox/{sandbox_dir}/bin/cdap sandbox start")
+   
+
+    # Build the plugin
+    os.chdir("plugin")
+
+    doc = xml.dom.minidom.parse("pom.xml")
+    modules = doc.getElementsByTagName("module")
+    modules_to_build : List[str] = []
+    if len(sys.argv) >= 1:
+        modules_to_build = sys.argv[1:]
+    print("arguments list: ", modules_to_build)
+    for moduleName in modules:
+        #getting module name
+        module = moduleName.firstChild.data
+        if module in modules_to_build:
+            print(f"Building plugin {module}")
+            run_shell_command(f"mvn clean package -pl {module} -am -DskipTests")
+            print (moduleName.firstChild.data)
+            # Get plugin artifact name and version from pom.xml.
+            os.chdir(f"{module}")
+            root = ET.parse('pom.xml').getroot()
+            plugin_name = root.find('{http://maven.apache.org/POM/4.0.0}artifactId').text
+            plugin_version = root.find('{http://maven.apache.org/POM/4.0.0}version').text
+            os.chdir("target")
+            plugin_properties = {}
+            plugin_parents = []
+            # Get plugin properties and parent from plugin json.
+            with open(f'{plugin_name}-{plugin_version}.json') as f:
+                obj = json.loads(f.read())
+                plugin_properties = obj['properties']
+                plugin_parents = obj['parents']
+
+            data = None
+            with open(f'{plugin_name}-{plugin_version}.jar', 'rb') as f:
+                data = f.read()
+
+            # Install the plugin on the sandbox.
+            print("Installing plugin")
+            res=requests.post(f"http://localhost:11015/v3/namespaces/default/artifacts/{plugin_name}", headers={"Content-Type": "application/octet-stream", "Artifact-Extends": '/'.join(plugin_parents), "Artifact-Version": plugin_version}, data=data)
+            assert res.ok or print(res.text)
+            res=requests.put(f"http://localhost:11015/v3/namespaces/default/artifacts/{plugin_name}/versions/{plugin_version}/properties", json=plugin_properties)
+            assert res.ok or print(res.text)
+            os.chdir("../..")
+               
+            # Run e2e tests
+            print("Preparing e2e framework")
+            os.chdir("../e2e")
```

Comment:
nit: can we combine Line 85 and Line 89?

Reply:
Something like this `os.chdir("../../../e2e")`

Reply:
done